framework/Shell/TShellInterpreter.php

```diff
@@ -0,0 +1,106 @@
+<?php
+/**
+ * TApplication class file
+ *
+ * @author Brad Anderson <belisoful@icloud.com>
+ * @link https://github.com/pradosoft/prado
+ * @license https://github.com/pradosoft/prado/blob/master/LICENSE
+ * @package Prado\Shell
+ */
+
+namespace Prado\Shell;
+
+use Prado\Prado;
+
+/**
+ * TShellInterpreter Class
+ *
+ * Command line interface, configures the action classes and dispatches the command actions.
+ *
+ * @author Wei Zhuo <weizhuo[at]gmail[dot]com>
+ * @package Prado\Shell
+ * @since 3.0.5
+ */
+class TShellInterpreter
```

Comment:
What do you think of making TShellInterpreter a subclass of TShellApplication?
It would reuse TShellApplication::_actionClasses related methods, creating the actual classes inside  TShellInterpreter::run()

Reply:
brilliant.

Reply:
I love the idea but there needs to be a directory specified by each command is not standardized.  This could get rid of some of the hacks for making application commands. the application could just MAKE a command without it being an app command.  The only way to get application specific commands is by having the directory to begin with.

Reply:
Got it